packages/token/src/tests/erc20/test_erc20.cairo

```diff
@@ -449,3 +451,135 @@ fn test_burn_from_zero() {
+
+    let mut spy = spy_events();
+    let contract_address = test_address();
+
+    start_cheat_caller_address(contract_address, RECIPIENT());
+
+    assert_state_before_mint(RECIPIENT());
+    state.update(ZERO(), RECIPIENT(), VALUE);
+    assert_state_after_mint(RECIPIENT(), VALUE);
+
+    spy.assert_only_event_transfer(contract_address, ZERO(), RECIPIENT(), VALUE);
+}
+
+#[test]
+fn test_update_from_zero_to_zero() {
+    // The update function should not be called from zero to zero
+    // because this does nothing to the state.
+    let mut state = setup();
+
+    let mut spy = spy_events();
+    let contract_address = test_address();
+    let supply_before_update = state.total_supply();
+
+    start_cheat_caller_address(contract_address, RECIPIENT());
+
+    state.update(ZERO(), ZERO(), VALUE);
+
+    let supply_after_update = state.total_supply();
+    assert_eq!(supply_before_update, supply_after_update);
+
+    spy.assert_only_event_transfer(contract_address, ZERO(), ZERO(), VALUE);
+}
+
+//
+// Helpers
+//
+
+fn assert_state_before_transfer(sender: ContractAddress, recipient: ContractAddress) {
+    let state = COMPONENT_STATE();
+    let initial_supply = SUPPLY;
+    let current_supply = state.total_supply();
+
+    assert_eq!(initial_supply, current_supply);
+    assert_eq!(state.balance_of(sender), SUPPLY);
+    assert!(state.balance_of(recipient).is_zero());
+}
+
+fn assert_state_after_transfer(sender: ContractAddress, recipient: ContractAddress, amount: u256) {
+    let state = COMPONENT_STATE();
+    let initial_supply = SUPPLY;
+    let current_supply = state.total_supply();
+
+    assert_eq!(initial_supply, current_supply);
+    assert_eq!(state.balance_of(sender), initial_supply - amount);
+    assert_eq!(state.balance_of(recipient), amount);
+}
+
+fn assert_state_before_mint(recipient: ContractAddress) {
+    let state = COMPONENT_STATE();
+    let initial_supply = SUPPLY;
+    let current_supply = state.total_supply();
+
+    assert_eq!(current_supply, initial_supply);
+    assert!(state.balance_of(recipient).is_zero());
+}
+
+fn assert_state_after_mint(recipient: ContractAddress, amount: u256) {
+    let state = COMPONENT_STATE();
+    let initial_supply = SUPPLY;
+    let current_supply = state.total_supply();
+
+    assert_eq!(state.balance_of(recipient), amount);
+    assert_eq!(current_supply, initial_supply + amount);
+}
+
+fn assert_state_before_burn(account: ContractAddress) {
+    let state = COMPONENT_STATE();
+    let initial_supply = SUPPLY;
+    let current_supply = state.total_supply();
+
+    assert_eq!(initial_supply, current_supply);
+    assert_eq!(state.balance_of(account), SUPPLY);
+}
+
+fn assert_state_after_burn(account: ContractAddress, amount: u256) {
+    let state = COMPONENT_STATE();
+    let initial_supply = SUPPLY;
+    let current_supply = state.total_supply();
+
+    assert_eq!(state.balance_of(account), initial_supply - amount);
+    assert_eq!(current_supply, initial_supply - amount)
+}
```

Comment:
Good job on adding helper functions. What do you think about adding messages to assert calls? I remember that we prefer to go without messages, but here each test case has at least 4 different assertions and messages inside helper functions could greatly simplify debugging

Reply:
That's a fair point. Done!